contracts/savings/SavingsContract.sol

```diff
@@ -743,4 +725,301 @@ contract SavingsContract is ISavingsContractV3, Initializable, InitializableToke
+        (shares, ) = _underlyingToCredits(assets);
+    }
+
+    /**
+     * @notice Burns enough vault shares from owner and transfers the exact amount of underlying asset tokens to the receiver.
+     * Emits a {Withdraw} event.
+     *
+     * @param assets The amount of underlying assets to be withdrawn from the vault.
+     * @param receiver The account that the underlying assets will be transferred to.
+     * @param owner Account that owns the vault shares to be burnt.
+     * @return shares The amount of vault shares that were burnt.
+     */
+    function withdraw(
+        uint256 assets,
+        address receiver,
+        address owner
+    ) external override returns (uint256 shares) {
+        require(assets > 0, "Must withdraw something");
+        uint256 _exchangeRate;
+        if (automateInterestCollection) {
+            ISavingsManager(_savingsManager()).collectAndDistributeInterest(address(underlying));
+        }
+        (shares, _exchangeRate) = _underlyingToCredits(assets);
+
+        _burnTransfer(assets, shares, receiver, owner, _exchangeRate, true);
+    }
+
+    /**
+     * @notice it must return a limited value if owner is subject to some withdrawal limit or timelock. must return balanceOf(owner) if owner is not subject to any withdrawal limit or timelock. MAY be used in the previewRedeem or redeem methods for shares input parameter. must NOT revert.
+     *
+     * @param owner Address that owns the shares.
+     * @return maxShares Total number of shares that owner can redeem.
+     */
+    function maxRedeem(address owner) external view override returns (uint256 maxShares) {
+        maxShares = balanceOf(owner);
+    }
+
+    /**
+     * @notice Allows an on-chain or off-chain user to simulate the effects of their redeemption at the current block, given current on-chain conditions.
+     *
+     * @return assets the exact amount of underlying assets that would be withdrawn by the caller if redeeming a given exact amount of Vault shares using the redeem method
+     */
+    function previewRedeem(uint256 shares) external view override returns (uint256 assets) {
+        (assets, ) = _creditsToUnderlying(shares);
+        return assets;
+    }
+
+    /**
+     * @notice Burns exact amount of vault shares from owner and transfers the underlying asset tokens to the receiver.
+     * Emits a {Withdraw} event.
+     *
+     * @param shares The amount of vault shares to be burnt.
+     * @param receiver The account the underlying assets will be transferred to.
+     * @param owner The account that owns the vault shares to be burnt.
+     * @return assets The amount of underlying assets that were transferred to the receiver.
+     */
+    function redeem(
+        uint256 shares,
+        address receiver,
+        address owner
+    ) external override returns (uint256 assets) {
+        require(shares > 0, "Must withdraw something");
```

Comment:
```suggestion
        require(shares > 0, "Must redeem something");
```

Reply:
This is the same error message as the old `redeemCredits` function so we'll just leave it with `Must withdraw something` in this ERC4626 implementation.